### PR TITLE
feat(cluster-dialer): cluster-dialer support record and query client data

### PR DESCRIPTION
**Pull request by chengjoey**
#### What this PR does / why we need it:
cluster-dialer support record and query client data

#### Specified Reviewers:

/assign @sfwn 

#### ChangeLog

| Language | Changelog |
| --------- | ------------ |
| 🇺🇸 English |    cluster-dialer support record and query client data          |
| 🇨🇳 中文    |   cluster-dialer支持记录查询客户端信息           |

#### Need cherry-pick to release versions?

Add comment like `/cherry-pick release/1.0` when this PR is merged.

> For details on the cherry pick process, see the [cherry pick requests](https://github.com/erda-project/erda/blob/master/CONTRIBUTING.md#how-to-cherry-pick-a-merged-pr) section under [CONTRIBUTING.md](https://github.com/erda-project/erda/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: apistructs/cluster_dialer.go [hunk 1]
```diff
@@ -47,3 +51,28 @@ func (c ClusterDialerClientType) MakeClientKey(clusterKey string) string {
...
+type ClusterDialerDataKey string
+
+var (
+	ClusterDialerDataKeyHost     ClusterDialerDataKey = "host"
+	ClusterDialerDataKeyPort     ClusterDialerDataKey = "port"
+	ClusterDialerDataKeyEndpoint ClusterDialerDataKey = "endpoint"
+	ClusterDialerDataKeyWorkerID ClusterDialerDataKey = "workerID"
+)
+
+type ClusterDialerClientData map[ClusterDialerDataKey]string
```
Comment (sfwn) [inline apistructs/cluster_dialer.go:64]: just a simple map is enough

Comment (sfwn) [reply apistructs/cluster_dialer.go:64]: define this specific key in client side

Comment (sfwn) [reply apistructs/cluster_dialer.go:64]: ClusterDialerClientDetail

File: apistructs/cluster_dialer.go [hunk 2]
```diff
@@ -47,3 +51,28 @@ func (c ClusterDialerClientType) MakeClientKey(clusterKey string) string {
...
+
+type ClusterDialerData map[string]ClusterDialerClientData
```
Comment (sfwn) [inline apistructs/cluster_dialer.go:66]: ClusterDialerClientMap

File: modules/cluster-dialer/server/tunnel-server.go [hunk 3]
```diff
@@ -174,9 +174,19 @@ func clusterRegister(server *remotedialer.Server, rw http.ResponseWriter, req *h
 			go registerFunc(clusterKey, clusterInfo)
 		}
 	default:
+		clientDataStr := req.Header.Get(apistructs.ClusterDialerHeaderKeyClientData.String())
+		if clientDataStr != "" {
+			var clientData apistructs.ClusterDialerClientData
+			if err := json.Unmarshal([]byte(clientDataStr), &clientData); err != nil {
+				logrus.Errorf("failed to unmarshal client data(skip clients update), err: %v", err)
```
Comment (sfwn) [inline modules/cluster-dialer/server/tunnel-server.go:181]: log more info

**Review by sfwn — CHANGES_REQUESTED** [review]

File: modules/cluster-dialer/server/tunnel-server.go (hunk 3 above)
Comment (chengjoey) [reply modules/cluster-dialer/server/tunnel-server.go]: added

File: apistructs/cluster_dialer.go (hunk 2 above)
Comment (chengjoey) [reply apistructs/cluster_dialer.go]: adjusted

File: apistructs/cluster_dialer.go (hunk 1 above)
Comment (chengjoey) [reply apistructs/cluster_dialer.go]: adjusted

File: apistructs/cluster_dialer.go [hunk 4]
```diff
@@ -47,3 +51,21 @@ func (c ClusterDialerClientType) MakeClientKey(clusterKey string) string {
 	}
 	return fmt.Sprintf("%s-client-type-%s", clusterKey, c)
 }
+
+type ClusterDialerDataKey string
```
Comment (sfwn) [inline apistructs/cluster_dialer.go:55]: not data key.
ClusterDialerClientDetailKey

**Review by sfwn — CHANGES_REQUESTED** [review]

File: apistructs/cluster_dialer.go (hunk 4 above)
Comment (chengjoey) [reply apistructs/cluster_dialer.go]: adjusted

**Review by sfwn — APPROVED** [review]

**Pull request merged**